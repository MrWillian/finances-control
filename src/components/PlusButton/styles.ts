import styled from 'styled-components/native';

export const Container = styled.TouchableOpacity`
  display: flex;
  align-items: center;
  justify-content: center;
  align-self: flex-end;

  background-color: #6269F1;

  width: 25px;
  height: 25px;
  margin: 10px 0;
  border-radius: 25px;
  elevation: 10;
`;
