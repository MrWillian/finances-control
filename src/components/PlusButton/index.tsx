import React from 'react';
import Icon from 'react-native-vector-icons/MaterialIcons';

import { Container } from './styles';

interface Props {
  onPress(): void;
}

const PlusButton: React.FC<Props> = ({ onPress }) => {
  return (
    <Container onPress={onPress}>
      <Icon name="add" size={20} color="#FFF" />
    </Container>
  );
}

export default PlusButton;