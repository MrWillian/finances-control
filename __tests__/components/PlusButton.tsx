import React from 'react';
import renderer from 'react-test-renderer';

import PlusButton from '../../src/components/PlusButton';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<PlusButton onPress={() => {}} />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
