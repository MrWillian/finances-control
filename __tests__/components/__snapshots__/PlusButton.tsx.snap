// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-end",
      "backgroundColor": "#6269F1",
      "borderRadius": 25,
      "display": "flex",
      "elevation": 10,
      "height": 25,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 0,
      "marginRight": 0,
      "marginTop": 10,
      "opacity": 1,
      "width": 25,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 20,
        },
        undefined,
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î……
  </Text>
</View>
`;
